### PR TITLE
chore(TS): migrate Point

### PR DESCRIPTION
**Pull request by ShaMan123**
nothing interesting

### CODE REVIEW

File: src/point.class.ts [hunk 1]
```diff
@@ -1,344 +1,348 @@
...
+  x: number
+
+  y: number
+
+  type = 'point'
+
+  static toPoint(from: IPoint) {
+    return new Point(from.x, from.y);
+  }
+
```
Comment (asturur) [inline src/point.class.ts:22]: is this new? what is it about?

**Comment by asturur** [comment]
multiplyEquals and divideEquals aren't available.
Now, i was thinking or we do full set of function that return a new point and mutate te point.

Or we do only return new.

File: src/point.class.ts (hunk 1 above)
Comment (asturur) [reply src/point.class.ts:22]: When @ShaMan123 says `nothing interesting` he is hiding a new motehod.

Comment (asturur) [reply src/point.class.ts:22]: Can't we have the class that can initialized both from 2 numbers or one object?
Is there a multiple pattern calls for constructors?

**Comment by ShaMan123** [comment]
I wanted to discuss that as well
I think no mutating is better. less errors in the long run
So dump all XXXEquals methods

File: src/point.class.ts (hunk 1 above)
Comment (ShaMan123) [reply src/point.class.ts:22]: good point!
on it

Comment (ShaMan123) [reply src/point.class.ts:22]: > When @ShaMan123 says `nothing interesting` he is hiding a new motehod.

Haha

**Review by ShaMan123 — COMMENTED** [review]
I removed the use of XXXEquals methods in fabric and marked them as deprecated
It was simple removing these methods which means they are not needed

**Review by asturur — APPROVED** [review]

**Pull request merged**